nio/service/base.py

```diff
@@ -148,10 +159,17 @@ def _execute_on_blocks_async(self, method):
         """
         threads = []
         for block in self._blocks.values():
-            threads.append(spawn(getattr(block, method)))
+            # no apparent way to retrieve block label from the thread object
+            threads.append({
```

Comment:
This is a good way to do it. Only suggestion would be to pas the block object rather than `block.name() or block.id()`. So like this:
```python
threads.append({
    "block": block,
    "thread": spawn(getattr(block, method))
})
...
raise BlockException(e, label=thread["block"].label())
```

Reply:
:+1: